clang/lib/Sema/SemaOpenMP.cpp

```diff
@@ -14633,6 +14633,26 @@ StmtResult Sema::ActOnOpenMPTargetTeamsDirective(ArrayRef<OMPClause *> Clauses,
   }
   setFunctionHasBranchProtectedScope();
 
+  bool HasBareClause = false;
+  bool HasThreadLimitClause = false;
+  bool HasNumTeamsClause = false;
+  OMPClause *BareClause = nullptr;
+
+  for (auto *C : Clauses) {
+    if (isa<OMPXBareClause>(C)) {
+      BareClause = C;
+      HasBareClause = true;
```

Comment:
Do you really need this flag? To just check that BareClause != nullptr is enough?

Reply:
It is not very necessarily needed, but I found it look nicer to check three boolean variables instead of two as well as a pointer. Just code preference. Nothing technical.